lmdeploy/pytorch/engine/engine_instance.py

```diff
@@ -137,25 +137,29 @@ async def async_stream_infer(self,
             sampling_param=sampling_param,
             adapter_name=adapter_name,
             input_multimodals=multimodal,
+            migration_request=gen_config.migration_request,
+            with_cache=gen_config.with_cache,
+            preserve_cache=gen_config.preserve_cache,
         )
         logger.debug(f'session[{session_id}] add message: num_input_ids={len(input_ids)}.')
         resp = self.req_sender.send_async(RequestType.ADD_MESSAGE, msg)
 
         while True:
             resp = await self.req_sender.async_recv(resp)
 
+            cache_block_ids = resp.data.get('cache_block_ids', None)
             if resp.type == ResponseType.SUCCESS:
-                token_ids = resp.data['token_ids'].tolist()
+                token_ids = resp.data['token_ids']
```

Comment:
EngineInstance would output ndarray instead of list[int], is it acceptable @lvhan028 ?

Reply:
No, it's not